incubator/hnc/pkg/reconcilers/hnc_config.go

```diff
@@ -136,28 +138,52 @@ func (r *ConfigReconciler) createObjectReconcilers(inst *api.HNCConfiguration) {
 	// from the forest for the object reconciliation, after we create ObjectReconcilers but before we write the
 	// ObjectReconcilers to the types list. As a result, objects of the newly added types might not be propagated correctly
 	// using the latest forest structure.
+	//
+	// For syncing an object reconciler, the mutex is guarding the read access to the `namespaces` field in the forest. The
+	// `namespaces` field is a shared resource between various reconcilers.
 	r.Forest.Lock()
 	defer r.Forest.Unlock()
 
 	for _, t := range inst.Spec.Types {
 		gvk := schema.FromAPIVersionAndKind(t.APIVersion, t.Kind)
-		if r.Forest.HasTypeSyncer(gvk) {
-			continue
+		mode := r.getMode(t)
+		if ts := r.Forest.GetTypeSyncer(gvk); ts != nil {
+			r.syncObjectReconciler(ctx, mode, ts)
+		} else {
+			r.createObjectReconciler(gvk, mode, inst)
+		}
+	}
+}
+
+// syncObjectReconciler syncs the mode of an ObjectReconciler. It also updates corresponding objects, if needed.
+func (r *ConfigReconciler) syncObjectReconciler(ctx context.Context, mode api.SynchronizationMode, ts forest.TypeSyncer) {
+	r.Log.Info("Syncing an object reconciler.", "gvk", ts.GetGVK())
```

Comment:
Nit: we generally don't have periods at the end of the messages, I don't think.

Reply:
Thanks for the suggestion. Removed all periods in all log messages in this file.